aries/aries_vcx/src/handlers/out_of_band/receiver.rs

```diff
@@ -62,11 +84,45 @@ impl OutOfBandReceiver {
         self.oob.clone().into()
     }
 
-    pub fn from_string(oob_data: &str) -> VcxResult<Self> {
-        Ok(Self {
-            oob: serde_json::from_str(oob_data)?,
-        })
+    pub fn to_json_string(&self) -> String {
+        self.to_aries_message().to_string()
     }
+
+    pub fn to_base64_url(&self) -> String {
+        BASE64_URL_SAFE.encode(self.to_json_string())
+    }
+
+    pub fn to_url(&self, domain_path: &str) -> VcxResult<Url> {
+        let mut oob_url = Url::parse(domain_path)?;
+        let oob_query = "oob=".to_owned() + &self.to_base64_url();
+        oob_url.set_query(Some(&oob_query));
+        Ok(oob_url)
+    }
+}
+
+fn from_json_string(oob_json: &str) -> VcxResult<Invitation> {
+    Ok(serde_json::from_str(oob_json)?)
+}
+
+fn from_base64_url(base64_url_encoded_oob: &str) -> VcxResult<String> {
+    Ok(String::from_utf8(
+        URL_SAFE_LENIENT.decode(base64_url_encoded_oob)?,
+    )?)
+}
+
+fn from_url(oob_url_string: &str) -> VcxResult<String> {
```

Comment:
nit; and general nit; IMO some of these function names are not obvious on what they are returning, i didn't know until i read into them. for instance, could you consider calling this fn something like `extract_encoded_invitation_from_url`.

Reply:
Ahh I agree--I would have preferred to keep them as methods on the Receiver, but since they return the initial instance they can't call each other (which is needed here), so I made them separate functions, but that resulted in some lost context. I'll adjust the function names to be more descriptive. :+1:

Reply:
ahh yea this context makes more sense